presto/eval.py

```diff
@@ -80,7 +81,7 @@ def finetune_sklearn_model(
         models: List[str] = ["Regression", "Random Forest"],
     ) -> Union[Sequence[BaseEstimator], Dict]:
         for model_mode in models:
-            assert model_mode in ["Regression", "Random Forest"]
+            assert model_mode in ["Regression", "Random Forest", "CatBoostClassifier"]
```

Comment:
A bit confused here. `CatBoostClassifier` is not a `BaseEstimator`, but it's here under `finetune_sklearn_model`. Would it make sense at some point to be future proof that different kinds of classifiers are wrapped inside a `BaseEstimator` (in the end, WorldCereal phase I code also had a `WorldCerealModel` base class to which sklearn, keras, catboost model classes were mapped for consistency. Just wondering how we can keep this section future-proof.

Reply:
This makes sense! I added an `SklearnStyleModel` class, which we can use for now.

Its not ideal because the CatBoost predict function behaves differently from the Sklearn ones, but I agree that this definitely makes it clearer.